src/coreclr/jit/lower.cpp

```diff
@@ -4242,6 +4274,57 @@ GenTree* Lowering::OptimizeConstCompare(GenTree* cmp)
             cmp->SetOperRaw(GenTree::ReverseRelop(cmp->OperGet()));
         }
 
+#ifdef TARGET_RISCV64
+        if (op2Value == 0 && !andOp2->isContained() && tryReduceSingleBitTestOps(op1->AsOp()))
+        {
+            GenTree* testedOp   = op1->gtGetOp1();
+            GenTree* bitIndexOp = op1->gtGetOp2();
+
+            if (bitIndexOp->IsIntegralConst())
+                bitIndexOp->SetContained();
+
+            LIR::Use cmpUse;
+            bool     isUserJtrue = BlockRange().TryGetUse(cmp, &cmpUse) && cmpUse.User()->OperIs(GT_JTRUE);
```

Comment:
What is this kind of contextual checking accomplishing? It is almost always an anti pattern. Why is the optimization not done when `JTRUE` is processed?